File: app/src/main/java/org/lantern/app/activity/SettingsActivity.java
package org.lantern.app.activity;

import android.content.res.Resources;
import androidx.fragment.app.FragmentActivity;
import android.text.Html;
import android.widget.CompoundButton;
import android.widget.TextView;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.ViewById;

import com.kyleduo.switchbutton.SwitchButton;

import org.lantern.app.LanternApp;
import org.lantern.mobilesdk.Logger;
import org.lantern.app.model.SessionManager;
import org.lantern.app.R;

@EActivity(R.layout.settings)
public class SettingsActivity extends FragmentActivity {
    private static final String TAG = SettingsActivity.class.getName();
    private static final String proxyAllDescFmt = "&#8226; %s<br />%s<br />&#8226; %s<br />%s";
    private static final SessionManager session = LanternApp.getSession();

    @ViewById
    TextView proxyAllDesc;

    @ViewById
    SwitchButton proxyAll;

    @AfterViews
    void afterViews() {
        if (session.proxyAll()) {
            proxyAll.setCheckedImmediatelyNoEvent(true);
            proxyAll.setBackColorRes(R.color.setting_on);
        }

        setDescText();

        proxyAll.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                proxyAll(isChecked);
            }
        });
    }

    private void setDescText() {
        final Resources res = getResources();
        proxyAllDesc.setText(Html.fromHtml(String.format(
            proxyAllDescFmt, res.getString(R.string.proxy_all_on_header),
            res.getString(R.string.proxy_all_on),
            res.getString(R.string.proxy_all_off_header),
            res.getString(R.string.proxy_all_off)
        )));
    }

    public void proxyAll(boolean on) {
        // store updated user preference
        session.setProxyAll(on);

        proxyAll.setBackColorRes(on ? R.color.setting_on : R.color.setting_off );

        Logger.debug(TAG, "Proxy all setting is " + on);
    }
}

File: android/app/src/main/java/org/getlantern/mobilesdk/activity/SettingsActivity.kt
package org.getlantern.mobilesdk.activity

import android.os.Bundle
import android.text.Html
import androidx.fragment.app.FragmentActivity
import org.getlantern.lantern.LanternApp
import org.getlantern.lantern.R
import org.getlantern.lantern.databinding.SettingsBinding
import org.getlantern.mobilesdk.Logger
import org.getlantern.mobilesdk.util.getStringWithAppName

open class SettingsActivity() : FragmentActivity() {
    private lateinit var binding: SettingsBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = SettingsBinding.inflate(layoutInflater)
        setContentView(binding.root)

        if (LanternApp.getSession().proxyAll()) {
            binding.proxyAll.setCheckedImmediatelyNoEvent(true)
            binding.proxyAll.setBackColorRes(R.color.setting_on)
        }
        setDescText()
        binding.proxyAll.setOnCheckedChangeListener { _, isChecked -> proxyAll(isChecked) }
    }

    private fun setDescText() {
        binding.proxyAllDesc.text = Html.fromHtml(String.format(
                proxyAllDescFmt, getString(R.string.proxy_all_on_header),
                getStringWithAppName(R.string.proxy_all_on),
                getString(R.string.proxy_all_off_header),
                getStringWithAppName(R.string.proxy_all_off)))
    }

    fun proxyAll(on: Boolean) {
        // store updated user preference
        LanternApp.getSession().setProxyAll(on)
        binding.proxyAll.setBackColorRes(if (on) R.color.setting_on else R.color.setting_off)
        Logger.debug(TAG, "Proxy all setting is $on")
    }

    companion object {
        private val TAG = SettingsActivity::class.java.name
        private const val proxyAllDescFmt = "&#8226; %s<br />%s<br />&#8226; %s<br />%s"
    }
}